consensus/src/member.rs

```diff
@@ -135,6 +136,66 @@ impl<D: Data, DP: DataProvider<D>, FH: FinalizationHandler<D>, US: Write, UL: Re
     }
 }
 
+struct MemberStatus<'a, H, D, S>
+where
+    H: Hasher,
+    D: Data,
+    S: Signature,
+{
+    pub task_queue: &'a BinaryHeap<ScheduledTask<H, D, S>>,
+    pub not_resolved_parents: &'a HashSet<H::Hash>,
+    pub not_resolved_coords: &'a HashSet<UnitCoord>,
+}
+
+impl<'a, H, D, S> fmt::Display for MemberStatus<'a, H, D, S>
+where
+    H: Hasher,
+    D: Data,
+    S: Signature,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let mut count: (usize, usize, usize, usize) = (0, 0, 0, 0);
+        for task in self.task_queue.iter().map(|st| &st.task) {
+            match task {
+                Task::CoordRequest(_) => count.0 += 1,
+                Task::ParentsRequest(..) => count.1 += 1,
+                Task::UnitMulticast(_) => count.2 += 1,
+                Task::RequestNewest(_) => count.3 += 1,
+            }
+        }
+        let pending_tasks: Vec<_> = self
+            .task_queue
+            .iter()
+            .filter(|st| match st.task {
+                Task::UnitMulticast(_) => false,
+                _ => st.counter >= 1,
+            })
+            .collect();
+        write!(f, "Member status report: ")?;
+        write!(
+            f,
+            "not_resolved_coords.len() - {}",
+            self.not_resolved_coords.len()
+        )?;
+        write!(
+            f,
+            "; not_resolved_parents.len() - {}",
+            self.not_resolved_parents.len()
+        )?;
+        write!(f, "; task queue content: ")?;
+        write!(
+            f,
+            "CoordRequest - {}, ParentsRequest - {}, UnitMulticast - {}, RequestNewest - {}",
+            count.0, count.1, count.2, count.3
+        )?;
+        if !pending_tasks.is_empty() {
+            write!(f, "; pending tasks - {:?}", pending_tasks)?;
```

Comment:
Maybe we'd leave it and increase the counter lower bound in line 171? The idea was to show/count tasks that spend too much time in the queue.

Reply:
Ok, you can probably do that (`counter > 5` ?), but if you choose so, then:
1. The `scheduled_time` display is pretty useless in the form `Instant { tv_sec: 1974910, tv_nsec: 936698995 }` -- the best fix is probably to not display this field at all.
2. `pending tasks` could be changed to something like `long time pending tasks` to describe the intention better.

Reply:
so how should I approach this? I'd like to implement `Display` for `ScheduledTask`, which prints the counter value and `Debug`-prints the inner Task, and then print all `ScheduledTask`s in a loop (with a trailing comma, because convenience...) - is this ok?

Reply:
@DamianStraszak approved, fixed